nixos/modules/services/web-apps/netbox.nix

```diff
@@ -0,0 +1,249 @@
+            # 'SENTINEL_SERVICE': 'netbox',
+            'PASSWORD': "",
+            'DATABASE': 0,
+            'SSL': False,
+            # Set this to True to skip TLS certificate verification
+            # This can expose the connection to attacks, be careful
+            # 'INSECURE_SKIP_TLS_VERIFY': False,
+        },
+        'caching': {
+            'HOST': 'localhost',
+            'PORT': 6379,
+            # Comment out `HOST` and `PORT` lines and uncomment the following if using Redis Sentinel
+            # 'SENTINELS': [('mysentinel.redis.example.com', 6379)],
+            # 'SENTINEL_SERVICE': 'netbox',
+            'PASSWORD': "",
+            'DATABASE': 1,
+            'SSL': False,
+            # Set this to True to skip TLS certificate verification
+            # This can expose the connection to attacks, be careful
+            # 'INSECURE_SKIP_TLS_VERIFY': False,
+        }
+    }
+
+    # This key is used for secure generation of random numbers and strings. It must never be exposed outside of this file.
+    # For optimal security, SECRET_KEY should be at least 50 characters in length and contain a mix of letters, numbers, and
+    # symbols. NetBox will not run without this defined. For more information, see
+    # https://docs.djangoproject.com/en/stable/ref/settings/#std:setting-SECRET_KEY
+    from os import environ, path
+    SECRET_KEY = open(path.join(environ.get("CREDENTIALS_DIRECTORY"), "netbox.secret")).read()
+  '';
+in
+
+{
+  options.services.netbox = {
+    enable = mkEnableOption "NetBox, an infrastructure resourece modelling application";
+
+    hostname = mkOption {
+      type = types.str;
+      example = "netbox.example.com";
+      description = ''
+        Hostname under which to serve the NetBox instance.
+      '';
+    };
+
+    secretFile = mkOption {
+      type = types.path;
+      default = "/var/lib/netbox/secret";
+      example = /run/keys/netbox.secret;
+      description = ''
+        Path to the secret used for secure random number and string
+        generation. It should be at least 50 characaster in length and
+        consist of a mix of letters, nummbers and symbols.
+
+        Will be auto-generated at the given path if it does not exist.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.redis.enable = true;
+
+    services.postgresql = {
+      enable = true;
+      ensureDatabases = [
+        "netbox"
+      ];
+      ensureUsers = [ {
+        name = "netbox";
+        ensurePermissions = {
+          "DATABASE netbox" = "ALL PRIVILEGES";
+        };
+      } ];
+    };
+
+    services.nginx = {
+      enable = true;
+      recommendedProxySettings = true;
+      virtualHosts."${cfg.hostname}" = {
+        location."/static" = {
+          alias = "${package}/opt/netbox/static";
+        };
+        location."/" = {
+          proxyPass = "http://unix:/run/netbox/netbox.sock:/";
+        };
+      };
+    };
```

Comment:
This is a simple nginx site that requires setting `cfg.hostname` which is also reused to harden `ALLOWED_HOSTS`.